gordon/router.py

```diff
@@ -142,13 +172,22 @@ def _get_next_phase(self, event_msg):
                 msg = f'Adding message "{event_msg}" to success channel.'
                 event_msg.append_to_history(msg, event_msg.phase)
                 await self.success_channel.put(event_msg)
+            elif old_phase == 'publish':
+                # if we're here, can assume message successfully went thru
+                # the entire phase route
```

Comment:
If the message is at this point, that means it originally came from `publish`, it successfully went through the  `cleanup` phase. Because the only two valid happy paths are `consume` -> `enrich` -> `publish` -> `cleanup` and `consume` -> `publish` -> `cleanup`, it is at this point that we can assume it never got dropped for some error, and it successfully completed the designated path. Does that make sense?